pkg/provider/vault/vault.go

```diff
@@ -0,0 +1,388 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vault
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
```

Comment:
not sure if you need to care yet, but the ioutil pkg is deprecated as of go 1.16: https://golang.org/doc/go1.16#ioutil

Reply:
Good catch, but not sure this can be updated until build pipeline is updated.